pyomeca/markers.py

```diff
@@ -1,180 +1,513 @@
+            ```
+
+            If you know the channel names, you can retrieve only the ones you are interested in by specifying strings:
+
+            ```python
+            channels = ["STER", "STERl"]
+            markers = Markers.from_trc(data_path, usecols=channels)
+            ```
+
+            Or by position:
+
+            ```python
+            channels = [3, 4]
+            markers = Markers.from_trc(data_path, usecols=channels)
+            ```
         """
-        Parameters
-        ----------
-        rt : RotoTrans
-            Rototranslation matrix to rotate about
-
-        Returns
-        -------
-        A new Vectors3d rotated
+        return read.read_trc(cls, filename, **kwargs)
+
+    @classmethod
+    def from_rototrans(cls, markers: xr.DataArray, rt: xr.DataArray) -> xr.DataArray:
         """
-        s_m = self.shape
-        s_rt = rt.shape
-
-        if len(s_rt) == 2 and len(s_m) == 2:
-            m2 = rt.dot(self)
-        elif len(s_rt) == 2 and len(s_m) == 3:
-            m2 = np.einsum("ij,jkl->ikl", rt, self)
-        elif len(s_rt) == 3 and len(s_m) == 3:
-            m2 = np.einsum("ijk,jlk->ilk", rt, self)
-        else:
-            raise ValueError("Size of RT and M must match")
+        Rotates markers data from a rototrans matrix.
 
-        return Markers3d(data=m2)
+        Arguments:
+            markers: markers array to rotate
+            rt: Rototrans to rotate about
 
-    def norm(self):
-        """
-        Compute the norm of the markers vector
+        Returns:
+            A rotated `xarray.DataArray`
+
+        !!! example
+            To rotate a random markers set from random angles:
+
+            ```python
+            from pyomeca import Angles, Rototrans, Markers
 
-        Parameters
-        ----------
-        Returns
-        -------
-        FrameDependentNpArray
+            n_frames = 100
+            n_markers = 10
+
+            angles = Angles.from_random_data(size=(3, 1, n_frames))
+            rt = Rototrans.from_euler_angles(angles, "xyz")
+            markers = Markers.from_random_data(size=(3, n_markers, n_frames))
+
+            rotated_markers = Markers.from_rototrans(markers, rt)
+            ```
         """
-        return np.linalg.norm(self[0:3, :, :], axis=0, keepdims=True)
+        return markers_from_rototrans(markers, rt)
+
+    @staticmethod
+    def _reshape_flat_array(array: Union[np.array, np.ndarray]) -> xr.DataArray:
+        if array.shape[1] % 3 != 0:
+            raise IndexError(
+                "Array second dimension should be divisible by 3. "
+                f"You provided an array with this shape {array.shape}"
+            )
+        return array.T.reshape((3, int(array.shape[1] / 3), array.shape[0]), order="F")
+
+    @staticmethod
+    def _get_requested_channels_from_pandas(
+        columns, header, usecols, prefix_delimiter: str, suffix_delimiter: str
+    ) -> Tuple[Optional[list], Optional[list]]:
```

Comment:
Is this duplicated from the other one?

Reply:
`Analogs` and `Markers` both have a different method `_get_requested_channels_from_pandas`.